src/training/clatter.rs

```diff
@@ -0,0 +1,48 @@
+use crate::common::{is_in_clatter, is_operation_cpu, is_training_mode, MENU};
+use smash::app::lua_bind::{ControlModule, EffectModule};
+use smash::app::BattleObjectModuleAccessor;
+use smash::lib::lua_const::*;
+use smash::phx::{Hash40, Vector3f};
+
+static mut COUNTER: u32 = 0;
+
+unsafe fn do_clatter_input(module_accessor: &mut BattleObjectModuleAccessor) {
+    ControlModule::add_clatter_time(module_accessor, -8.0, 0);
+    let zeros = Vector3f {
+        x: 0.0,
+        y: 0.0,
+        z: 0.0,
+    };
+
+    EffectModule::req_on_joint(
+        module_accessor,
+        Hash40::new("sys_clatter"),
+        Hash40::new("hip"),
+        &zeros,
+        &zeros,
+        1.0,
+        &zeros,
+        &zeros,
+        true,
+        *EFFECT_SUB_ATTRIBUTE_NO_JOINT_SCALE as u32
+            | *EFFECT_SUB_ATTRIBUTE_FOLLOW as u32
+            | *EFFECT_SUB_ATTRIBUTE_CONCLUDE_STATUS as u32,
+        0,
+        0,
+    );
+}
+
+pub unsafe fn handle_clatter(module_accessor: &mut BattleObjectModuleAccessor) {
+    if !is_training_mode() || !is_operation_cpu(module_accessor) {
+        return;
+    }
+
+    if !is_in_clatter(module_accessor) {
+        return;
+    }
+
+    COUNTER = (COUNTER + 1) % MENU.clatter_strength.into_u32();
```

Comment:
Since we're using u32::max for None, shouldn't we special case that here? Otherwise we will eventually hit this.

Reply:
I don't think it's necessary. u32::max frames is 828 days. If there's one clatter input in that timeframe, I think we're okay

Reply:
Though this did make me realize that even using u32::max would do one clatter input on the second frame. I changed the logic condition to be `if COUNTER == setting-1` instead of `IF COUNTER == 1`